### PR TITLE
Add kafka watcher

### PR DESCRIPTION
**Pull request by craigbr**
Uses monasca/monasca-watchers. Creates a container for kafka-watcher

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -69,6 +69,14 @@ services:
     restart: on-failure
     depends_on:
       - zookeeper
+  watcher:
```
Comment (matrixik) [inline docker-compose.yml]: should be `kafka-watcher`

File: docker-compose.yml [hunk 2]
```diff
@@ -69,6 +69,14 @@ services:
...
+    image: kafka-watcher:${MON_KAFKA_WATCHER_VERSION}
+    environment:
```
Comment (matrixik) [inline docker-compose.yml]: Does it have any logs? If yes then add `LOGSTASH_FIELDS: "service=kafka-watcher"`

File: kafka-watcher/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+-------------
+
+This parameter can be specified using environment variables:
+
+| Variable                      | Default          | Description                           |
+|-------------------------------|------------------|---------------------------------------|
+| `STAY_ALIVE_ON_FAILURE`       | `false`          | If `true`, container stays alive for 2 hours after kafka watcher exits |
+
+
+Additionaly parameters are available.  See https://github.com/craigbr/monasca-watchers for more details.
```
Comment (matrixik) [inline kafka-watcher/README.md]: Remove one space before `See`

Comment (matrixik) [inline kafka-watcher/README.md]: Should be https://github.com/monasca/monasca-watchers (`monasca` organization)

File: kafka-watcher/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# `docker build`
+ARG REBUILD=1
+
+ARG WATCHER_REPO=https://github.com/monasca/monasca-watchers
+ARG WATCHER_BRANCH=master
+
+ENV GOPATH=/go
+ENV PROJECT_DIR=$GOPATH/src/github.com/monasca/monasca-watchers
+
+RUN apk add --no-cache openssl py2-jinja2
```
Comment (matrixik) [inline kafka-watcher/Dockerfile]: I know it wont be added to the end image but what for you install `py2-jinja2`?

File: kafka-watcher/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+RUN apk add --no-cache --virtual build-dep librdkafka-dev \
+      git go glide make g++ openssl-dev
```
Comment (matrixik) [inline kafka-watcher/Dockerfile]: Why installing `make` and `openssl` in line 14 if you install `openssl-dev` here?

File: kafka-watcher/Dockerfile (hunk 4 above)
Comment (craigbr) [reply kafka-watcher/Dockerfile]: Removed.

File: kafka-watcher/Dockerfile (hunk 5 above)
Comment (craigbr) [reply kafka-watcher/Dockerfile]: Removed. Thanks

**Comment by craigbr** [comment]
@matrixik I believe I have addressed your comments? Can you remove your "requested changes" if you agree? Thanks for the review.

**Comment by matrixik** [comment]
I merged it but it's failing to build:
```
[WARN]	Unable to checkout github.hpe.com/kronos/kelog
[ERROR]	Update failed for github.hpe.com/kronos/kelog: Cannot detect VCS

[ERROR]	Failed to install: Cannot detect VCS
```
Personally I also can't access github.hpe.com/kronos/kelog

**Comment by timothyb89** [comment]
Looks like that issue was fixed in https://github.com/monasca/monasca-watchers/commit/03f01e494c5e76a0c7dd3566d3e751aef41295e2

**Comment by matrixik** [comment]
Yes, it was rebuilded in https://github.com/monasca/monasca-docker/pull/348

**Comment by matrixik** [comment]
@timothyb89 You was right that it still is not working with docker-compose.yml file because image URI was wrong. Fixed in https://github.com/monasca/monasca-docker/pull/356

**Pull request merged**